### DISCUSSION TITLE
Why this isn't a good tool for developers and how to fix some of that

### DISCUSSION THREAD

**Discussion by systemlord01** [Pythagora-io/gpt-pilot#804] [Ideas]
So I am a senior software developer. Not an AI expert. And I am going to tell you why I got so frustrated I had to cancel the subscription and went back to using DeepSeek for suggesting some code for specific problems.

The problem is the workflow does NOT match the workflow of a developer.

It's all good to have a plan of action at the beginning of a project. But as you go forward stuff changes. Previous decisions have to be overthrown. Project scope changes ect ... 

And this is where this tool completely fails. It makes up the whole plan and tries to implement it all by itself in one go. His way. And cannot stop. It tries to troubleshoot bugs BEFORE even asking if I know the solution. It does NOT remember when I point out a mistake (like executing commands in the wrong folder, FOR THE BILIONTH TIME FRONTEND CODE IS IN THE FRONTEND FOLDER, REMEMBER!!!!).

In short instead of having a sequential step by step process do the following.

Let it make a backlog. You can then decide which items on the backlog you want to pick up and in which order. You can decide to remove or add items to the backlog. This could be done by just describing and it can the propose a list of tickets and what tickets to remove. When implementing a ticket it can suggest code, but then again ask for feedback, and this should go back and forth.  

You have to create the feel of a peer programmer. Not an autistics developer with his own plan not listening to anybody cruising full speed ahead towards an iceberg.

Don't get me wrong it's very impressive. But as it stands now a massive waist of time to get it to do what you want it to do and keeping it from completely fucking up your project because you forgot to commit.

I think using the backlog system. Constant feedback, ect ... would make it a lot better. Also the ability to explain to it what the project structure is and it remembering it would be nice. This was not my experience.

Looking through the docs this whole thing seems to be the cause of my troubles:
https://github.com/Pythagora-io/gpt-pilot/wiki/Command-Execution

**Comment by Mickelodian** [comment]
Okay so I was about to try this Pytagora now that they have funding, and therefore a good reason not to screw up! In fact set up openrouter in the .env file. Created a budget, a relatively simple app to build. It never even got that far. 

After installing it in VS Code, as a first bad omen I got "Error: spawn e:\AI\CODEBASES\AGSTUDIO\GPTPilot\gpt-pilot\pilot-env\Scripts\python ENOENT"

Thats not a good thing to display to any user on their first go... thats before it even started to ask any questions. Like, no chance to ask the agent anything at all. It screwed up right out of the gate when launched! Also its not coming as a major surprise that the app can't initiate the scripts in that folder since that folder literally does not exist anyway! Tells me from the get go "sloppy and careless code in this app" and I can't afford that! I have my own code to worry about without having the fear of someone else's software bringing my work to a halt. 
Now looking at your post I'd have to agree... if it cannot bend even slightly if I did get it working then what is the point? 

Pytagora is now off my consideration list before we even start! I'll get back to it if they survive the next 6 months. Same decision I made 3 months ago! When again I never even got to try it. 
Thanks for the outline of your experience.  Appreciated.

**Comment by skatekowski** [comment]
> So I am a senior software developer. Not an AI expert. And I am going to tell you why I got so frustrated I had to cancel the subscription and went back to using DeepSeek for suggesting some code for specific problems.
> 
> The problem is the workflow does NOT match the workflow of a developer.
> 
> It's all good to have a plan of action at the beginning of a project. But as you go forward stuff changes. Previous decisions have to be overthrown. Project scope changes ect ...
> 
> And this is where this tool completely fails. It makes up the whole plan and tries to implement it all by itself in one go. His way. And cannot stop. It tries to troubleshoot bugs BEFORE even asking if I know the solution. It does NOT remember when I point out a mistake (like executing commands in the wrong folder, FOR THE BILIONTH TIME FRONTEND CODE IS IN THE FRONTEND FOLDER, REMEMBER!!!!).
> 
> In short instead of having a sequential step by step process do the following.
> 
> Let it make a backlog. You can then decide which items on the backlog you want to pick up and in which order. You can decide to remove or add items to the backlog. This could be done by just describing and it can the propose a list of tickets and what tickets to remove. When implementing a ticket it can suggest code, but then again ask for feedback, and this should go back and forth.
> 
> You have to create the feel of a peer programmer. Not an autistics developer with his own plan not listening to anybody cruising full speed ahead towards an iceberg.
> 
> Don't get me wrong it's very impressive. But as it stands now a massive waist of time to get it to do what you want it to do and keeping it from completely fucking up your project because you forgot to commit.
> 
> I think using the backlog system. Constant feedback, ect ... would make it a lot better. Also the ability to explain to it what the project structure is and it remembering it would be nice. This was not my experience.
> 
> Looking through the docs this whole thing seems to be the cause of my troubles: https://github.com/Pythagora-io/gpt-pilot/wiki/Command-Execution

This is a perfect explanation of why Pythagora isn't worth using. So sad. I got stuck in the loop many times just because I couldn't do suggestions. If you suggest something it takes several rounds before "the developer" reacts. Having no backlog and a developer that just don't listens to the user or the client would be the dead of every dev company...

**Comment by blarvin** [comment]
I think #900 goes a long way towards solving this issue, and many others. 

If the Agents used a user-facing document as the source of truth, and the user could update that, then the User could better STEER the Agents. 

For instance, when the Dev Agent goes in the wrong direction, we should be able to clarify the direction we want by updating such a file and then the project should integrate that going forward.

This file could be very simple, just the original prompt. Or it could get more specific with a section on "STACK" including dependency versions. It could include for instance a section to define development "phases" such as "hello world phase", "stack integration phase", and "feature adding phase", which should keep the Agents from charging ahead on hallucinated plans.
